tests/manager/repositories/user_resource_policy/test_user_resource_policy_repository.py

```diff
@@ -1,265 +1,256 @@
-            assert hasattr(method, "__wrapped__") or hasattr(method, "__name__")
-
-    def test_repository_initialization(self, mock_db_engine):
-        """Test repository initialization"""
-        repo = UserResourcePolicyRepository(db=mock_db_engine)
-        assert repo._db == mock_db_engine
-
-    @pytest.mark.asyncio
-    async def test_create_with_none_values(self, repository, mock_db_engine):
-        """Test creating a policy with None values (should still work)"""
-        # Setup mock session and row
-        mock_session = AsyncMock()
-        mock_db_engine.begin_session.return_value.__aenter__.return_value = mock_session
-
-        policy_data_with_none = {
-            "name": "test-policy",
-            "max_vfolder_count": None,
-            "max_quota_scope_size": None,
-            "max_session_count_per_model_session": None,
-            "max_customized_image_count": None,
-        }
-
-        expected_result = UserResourcePolicyData(
-            name="test-policy",
-            max_vfolder_count=0,  # Database default
-            max_quota_scope_size=0,  # Database default
-            max_session_count_per_model_session=0,  # Database default
-            max_customized_image_count=3,  # Database default
-        )
-
-        # Mock the row creation and to_dataclass conversion
-        mock_row = MagicMock(spec=UserResourcePolicyRow)
-        mock_row.to_dataclass.return_value = expected_result
-
-        with patch(
-            "ai.backend.manager.repositories.user_resource_policy.repository.UserResourcePolicyRow"
-        ) as mock_row_class:
-            mock_row_class.return_value = mock_row
-
-            # Execute the method
-            result = await repository.create(policy_data_with_none)
-
-            # Verify the result
-            assert result == expected_result
-
-            # Verify the row was created with the data (including None values)
-            mock_row_class.assert_called_once_with(**policy_data_with_none)
+    async def test_create_and_get_roundtrip(self, repository: UserResourcePolicyRepository) -> None:
+        """Test creating a policy and retrieving it"""
+        policy_name = "test-policy-roundtrip"
+        max_vfolder_count = 15
+        max_quota_scope_size = 500000
+        max_session_count_per_model_session = 10
+        max_customized_image_count = 5
+
+        try:
+            created = await repository.create(
+                UserResourcePolicyCreator(
+                    name=policy_name,
+                    max_vfolder_count=max_vfolder_count,
+                    max_quota_scope_size=max_quota_scope_size,
+                    max_session_count_per_model_session=max_session_count_per_model_session,
+                    max_customized_image_count=max_customized_image_count,
+                    max_vfolder_size=None,  # Deprecated
+                )
+            )
+            retrieved = await repository.get_by_name(created.name)
```

Comment:
It seems better to write tests for the case where you've already created it with a fixture, and tests for the case where you haven't, rather than creating and removing it here.